### ISSUE TITLE
Long REPL inputs display extra copies when reaching the end of the line

### ISSUE THREAD

**Issue by tkelman** [JuliaLang/julia#6488]
Noticing this on Windows:
```
julia> versioninfo()
Julia Version 0.3.0-prerelease+2584
Commit 951e027* (2014-04-10 13:40 UTC)
Platform Info:
  System: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-2630QM CPU @ 2.00GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY)
  LAPACK: libopenblas
  LIBM: libopenlibm

julia> println("Typing a long input line at the REPL displays extra copies of th
julia> println("Typing a long input line at the REPL displays extra copies of th
julia> println("Typing a long input line at the REPL displays extra copies of th
e line when wrapping to the next line")
```

Or if I SSH into a Linux machine using either Putty or Cygwin, I see the same basic problem with Linux-Julia (same commit) but with only one extra copy instead of two.

**Comment by tkelman** [comment]
@davidanthoff you have a test vm of windows 10, right? Can you check if this issue is present there?

**Comment by davidanthoff** [comment]
Traveling this week, and then on vacation for two weeks... I can take a look once I'm back, if this is still an issue then.

**Comment by tkelman** [comment]
No rush, you reminded me about windows 10 in a different issue so figured it was worth checking this one there at some point.

**Comment by tkelman** [comment]
closed by #15491 but needs a test or it's going to regress

**Comment by vtjnash** [comment]
i'm not exactly sure how you would test this, since it is particular to the behavior of the output tty (the code in this function is an approximate model of how the attached tty is expected to behave)
